docs/_docs/04_community/client_factory.md

```diff
@@ -0,0 +1,176 @@
+    subscriptionID := "your subscription ID"
+
+    // Create a VM client and return
+    vmClient, err := compute.NewVirtualMachinesClient(subscriptionID)
+
+    // Use client / etc
+}
+```
+
+However, this code will not work in non-Public cloud environments as the REST endpoints have different URIs depending on environment.  Instead, you need to use an alternative method (provided in the Azure REST SDK for Go) to get a properly configured client(*all REST API clients should support this alternate method*):
+
+```golang
+import (
+    "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+)
+
+func SomeVMHelperMethod() {
+    subscriptionID := "your subscription ID"
+    baseURI := "management.azure.com"
+
+    // Create a VM client and return
+    vmClient, err := compute.NewVirtualMachinesClientWithBaseURI(baseURI, subscriptionID)
+
+    // Use client / etc
+}
+```
+
+Using code similar to above, you can communicate with any Azure cloud environment just by changing the base URI that is passed to the clients (Azure Public shown in above example).
+
+## Lookup Environment Metadata
+
+Developers MUST avoid hardcoding these base URI's.  Instead, they should be looked up from an authoritative source. The AutoRest-GO library (used by the Go SDK) provides such functionality. The `client_factory` module makes use of the AutoRest `EnvironmentFromName(envName string)` function to return the appropriate structure.  This method and Environment structure is documented on GoDoc [here](https://godoc.org/github.com/Azure/go-autorest/autorest/azure#EnvironmentFromName).
+
+To configure different cloud environments, we will use the same `AZURE_ENVIRONMENT` environment variable that the Go SDK uses. This can currently be set to one of the following values:
+
+|Value                      |Cloud Environment  |
+|---------------------------|-------------------|
+|"AzureChinaCloud"          |ChinaCloud         |
+|"AzureGermanCloud"         |GermanCloud        |
+|"AzurePublicCloud"         |PublicCloud        |
+|"AzureUSGovernmentCloud"   |USGovernmentCloud  |
+|"AzureStackCloud"          |Azure stack        |
+
+When using the "AzureStackCloud" setting, you MUST also set the `AZURE_ENVIRONMENT_FILEPATH` variable to point to a JSON file containing your Azure Stack URI details.
+
+## Putting it all together
+
+ `client_factory` implements this pattern described above in order to instantiate and return properly configured *REST SDK for GO* clients so that test implementers don't have to consider REST API client implementation as long as they have the correct `AZURE_ENVIRONMENT` env setting.  If this environment variable is not set, the client will assume public cloud as the cloud environment to communicate with.  We strongly recommend developers creating TerraTest helper methods for Azure use this pattern with client factory to create REST API clients.  This will reduce effort for TerraTest users creating test for Azure resources.
```

Comment:
```suggestion
 `client_factory` implements this pattern described above in order to instantiate and return properly configured *REST SDK for GO* clients so that test implementers don't have to consider REST API client implementation as long as they have the correct `AZURE_ENVIRONMENT` env setting.  If this environment variable is not set, the client will assume public cloud as the cloud environment to communicate with.  We strongly recommend developers creating Terratest helper methods for Azure use this pattern with client factory to create REST API clients.  This will reduce effort for Terratest users creating test for Azure resources.
```